### PR TITLE
Add links to source code

### LINKED ISSUE #41: Generate a link to source
Provide the ability to (optionally) generate a link to the source code that the documentation was generated for. For example:

# [Summary.Doc](https://github.com/candy-kingdom/summary/blob/c3996afb1df86bff168c0e9545d75a316244e022/src/Core/Doc.cs#L7C1-L7C1)
```cs
public record Doc(DocMember[] Members)
```

A document parsed from the source code or an assembly.

### [Empty](https://github.com/candy-kingdom/summary/blob/c3996afb1df86bff168c0e9545d75a316244e022/src/Core/Doc.cs#L12)
```cs
public static readonly Doc Empty
```

An empty document.

...

### PR DESCRIPTION
**Pull request by kostiantxn**
Closes #41.

### CODE REVIEW

File: src/Plugins/Markdown/MdRenderer.cs [hunk 1]
```diff
@@ -95,20 +99,41 @@ private MdRenderer Property(DocTypeDeclaration parent, DocProperty property)
...
+        };
+
+        string Link(string text)
+        {
+            if (member.Location is not null)
+            {
+                var from = new Uri(_output);
+                var to = new Uri(member.Location.Path);
+
+                return $"[{text}]({from.MakeRelativeUri(to)}#L{member.Location.Start.Line + 1})";
```
Comment (kostiantxn) [inline src/Plugins/Markdown/MdRenderer.cs:131]: The line number format should be configurable. For example, GitHub uses `#L3-L7`, while Bitbucket uses `#lines-3:7`.

Comment (kostiantxn) [inline src/Plugins/Markdown/MdRenderer.cs:131]: We should also consider adding columns to the URL (e.g., `#L3C5`).

Comment (joshua-light) [reply src/Plugins/Markdown/MdRenderer.cs:131]: What do you think about extracting a new `MdRendererOptions` class for tweaking such things?

For example, we could call the `UseMdRenderer` method on the pipeline with few ways:
(the old one)
```cs
pipeline
  .UseMdRenderer("...");
```
(the more customizable one)
```cs
pipeline
  .UseMdRenderer(options => options
    .CleanupOutputDir()
    .UseGitHubMarkdown())
```

**Review by joshua-light — APPROVED** [review]
This feature is busted!

File: src/Plugins/Markdown/MdRenderer.cs (hunk 1 above)
Comment (kostiantxn) [reply src/Plugins/Markdown/MdRenderer.cs:131]: I think it's a really good idea! I was also planning to raise a ticket to create a configuration mechanism for the pipeline, which I'll do shortly. 👍

**Pull request merged**